src/pages/EditScholarship.tsx

```diff
@@ -18,20 +18,24 @@ import Scholarships from '../models/Scholarships';
 import { Helmet } from 'react-helmet';
 import useAuth from '../lib/useAuth';
 import ScholarshipsContext from '../models/ScholarshipsContext';
+import ScholarshipData from '../types/ScholarshipData';
+import Model from '../models/base/Model';
 
-function EditScholarship() {
+function EditScholarship(): JSX.Element {
   const { id } = useParams();
-  const [scholarship, setScholarship] = useState(undefined);
-  const [error, setError] = useState();
-  const [delError, setDelError] = useState();
+  const [scholarship, setScholarship] = useState<
+    Model<ScholarshipData> | undefined
+  >(undefined);
+  const [error, setError] = useState<string>();
+  const [delError, setDelError] = useState<string>();
   const [dialogOpen, setDialogOpen] = useState(false);
   const navigate = useNavigate();
 
   // Fetch the scholarship
   useEffect(() => {
     let mounted = true;
 
-    Scholarships.id(id)
+    Scholarships.id(id!)
```

Comment:
what's the `!` for?

Reply:
In TypeScript this says "I assert this is not undefined!"